src/legacy/core_plugins/data/public/search/search_bar/components/search_bar.tsx

```diff
@@ -46,7 +47,7 @@ export interface SearchBarProps {
   appName: string;
   intl: InjectedIntl;
   uiSettings: UiSettingsClientContract;
-  savedQueryService: SavedQueryService;
+  savedObjectsClient: SavedObjectsClientContract;
```

Comment:
`savedObjectsClient` does not seem to change over time, thus it could be outside of `state`

```tsx
private savedQueryService = createSavedQueryService(this.props.savedObjectsClient);
```